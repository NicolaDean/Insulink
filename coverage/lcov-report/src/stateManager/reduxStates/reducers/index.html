
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/stateManager/reduxStates/reducers</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../../index.html">All files</a> src/stateManager/reduxStates/reducers</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.21% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>144/171</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">54.34% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>25/46</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">73.91% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>17/23</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">84.61% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>143/169</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="1" class="abs high">1/1</td>
	</tr>

<tr>
	<td class="file high" data-value="macroTracker.js"><a href="macroTracker.js.html">macroTracker.js</a></td>
	<td data-value="88.61" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 88%"></div><div class="cover-empty" style="width: 12%"></div></div>
	</td>
	<td data-value="88.61" class="pct high">88.61%</td>
	<td data-value="123" class="abs high">109/123</td>
	<td data-value="52.94" class="pct medium">52.94%</td>
	<td data-value="34" class="abs medium">18/34</td>
	<td data-value="80" class="pct high">80%</td>
	<td data-value="15" class="abs high">12/15</td>
	<td data-value="89.34" class="pct high">89.34%</td>
	<td data-value="122" class="abs high">109/122</td>
	</tr>

<tr>
	<td class="file medium" data-value="userReducer.js"><a href="userReducer.js.html">userReducer.js</a></td>
	<td data-value="72.34" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 72%"></div><div class="cover-empty" style="width: 28%"></div></div>
	</td>
	<td data-value="72.34" class="pct medium">72.34%</td>
	<td data-value="47" class="abs medium">34/47</td>
	<td data-value="58.33" class="pct medium">58.33%</td>
	<td data-value="12" class="abs medium">7/12</td>
	<td data-value="62.5" class="pct medium">62.5%</td>
	<td data-value="8" class="abs medium">5/8</td>
	<td data-value="71.73" class="pct medium">71.73%</td>
	<td data-value="46" class="abs medium">33/46</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at Thu May 26 2022 11:26:03 GMT+0200 (Central European Summer Time)
            </div>
        <script src="../../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../../sorter.js"></script>
        <script src="../../../../block-navigation.js"></script>
    </body>
</html>
    