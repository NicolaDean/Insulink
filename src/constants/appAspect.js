import { Dimensions  } from 'react-native';
// primary in hex: #70CAE6
export const themeBlue = {
    white:"#fff",
    black:"black",
    primary:"rgb(112,202,230)",
    secondary:"",

};

export const colors = themeBlue;


const dimension = Dimensions.get('window');

export const dim = {
    height: dimension.height,
    width : dimension.width,
}