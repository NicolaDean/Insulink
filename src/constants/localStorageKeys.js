
export const storageKeys ={
    dataAvailability: "AVAILABILITY",
    userData: "USER_DATA",
    foodDiary: "FOOD_DIARY",
    
}