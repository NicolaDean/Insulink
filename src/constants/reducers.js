export const foodMethods = {
    addFood:'ADD_FOOD',
    removeFood:'REMOVE_FOOD',
    selectMeal:'SELECT_MEAL',
}

export const userMethods = {
    registerUser: 'REGISTER_USER',
    deleteUser: 'DELETE_USER',
    logout: 'LOGOUT_USER',
    login: 'LOGIN_USER',
    updateUser: 'UPDATE_USER',
    addGlicemy: 'ADD_GLICEMY',
}