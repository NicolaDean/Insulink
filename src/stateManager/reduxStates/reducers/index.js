import macroReducer from "./macroTracker";
import { combineReducers } from "redux";

const appReducers = combineReducers({
    macroTracker : macroReducer,
});

export default appReducers;