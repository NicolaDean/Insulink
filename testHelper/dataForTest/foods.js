
export const dummyAppleDeletable = {"id":{"food_name":"apple pie","serving_unit":"piece (1/8 of 9\" dia)","tag_name":"Apple pie","serving_qty":1,"common_type":null,"tag_id":"15060","photo":{"thumb":"https://nix-tag-images.s3.amazonaws.com/1291_thumb.jpg"},"locale":"en_US"},"name":"apple pie","image":"https://nix-tag-images.s3.amazonaws.com/1291_highres.jpg","cal":237,"carb":34,"fat":11,"prot":1.9,"quantity":1,"unit":"g","identifier":1}

export const dummyApple = {"food_name":"apple pie","serving_unit":"piece (1/8 of 9\" dia)","tag_name":"Apple pie","serving_qty":1,"common_type":null,"tag_id":"15060","photo":{"thumb":"https://nix-tag-images.s3.amazonaws.com/1291_thumb.jpg"},"locale":"en_US"}
