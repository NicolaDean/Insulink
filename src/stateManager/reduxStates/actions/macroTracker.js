import { foodMethods } from "../../../constants/reducers"

export const addFood = (food) =>{
    return{
        type: foodMethods.addFood,
        payload: {food:food}
    }
} 

export const removeFood = (food,mealType) =>{
    return{
        type: foodMethods.removeFood,
        payload: {food:food,mealType:mealType}
    }
} 

export const selectMealType = (mealType) =>{
    return{
        type: foodMethods.selectMeal,
        payload: {mealType:mealType}
    }
}