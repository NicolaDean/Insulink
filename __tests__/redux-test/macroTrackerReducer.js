import React from 'react';
import { foodMethods, userMethods } from '../../src/constants/reducers';
import macroTracker, { initialDiaryState } from '../../src/stateManager/reduxStates/reducers/macroTracker';
import { dummyApple } from '../../testHelper/dataForTest/foods';

const initState = {...initialDiaryState,currentDate:"[TODAY DATE]" };

const food_test={
    id:     dummyApple,
    name:   "apple",
    image:  "[IMAGE]",
    cal:    90 ,
    carb:   33,
    fat:    55,
    prot:   22,
    quantity: 1,
    unit: 'g',
    identifier: 1,
}

const food_test2={
    id:     dummyApple,
    name:   "apple",
    image:  "[IMAGE]",
    cal:    180 ,
    carb:   55,
    fat:    33,
    prot:   77,
    quantity: 2,
    unit: 'g',
    identifier: 1,
}

let dummy_sport = {
    name:"Nome",
    calories:400,
    duration:50,
    photo:"[IMAGE]",
    identifier:1
}

const fir = ()=>{
    return {
      doc:null,
      collection: (a) =>{return "ciao"},
    };
}

jest.mock("@react-native-firebase/firestore", () => fir);


/**
 * Those test allow to check if the reducer change correctly the global state after an action
 */

describe("TESTING THE REDUX REDUCER FOR MACRO:", ()=>{


    beforeAll(()=>{

    });

    afterAll(()=>{

    });


    test("Add,Edit,Remove Food",()=>{
        console.log(initState);
        const action ={
            type:foodMethods.addFood,
            payload:{
                food:food_test
            }
        }
        let prevState; //State of previous operation
        //TRY ADDING A FOOD
        expect(prevState = macroTracker(initState,action)).toMatchSnapshot();

        //EDITING THE FOOD
        action.type = foodMethods.editFood;
        action.payload = {food:food_test2};
        expect(prevState = macroTracker(prevState,action)).toMatchSnapshot();

        //REMOVING THE FOOD
        action.type = foodMethods.removeFood;
        expect(macroTracker(prevState,action)).toMatchSnapshot();
    });

    test("Add,Remove  Sport",()=>{
        const action ={
            type:foodMethods.addActivity,
            payload:{
                activity:dummy_sport
            }
        }
        let prevState; //State of previous operation
        //TRY ADDING A SPORT
        expect(prevState = macroTracker(initState,action)).toMatchSnapshot(
            {
                currentDate: expect.any(String),
            }      
        );

        //TRY REMOVING A SPORT
        action.type = foodMethods.removeActivity;
        expect(prevState = macroTracker(prevState,action)).toMatchSnapshot()
    });


    test("Select Meal",()=>{

    });

    test("Reset Diary",()=>{
        const action ={
            type:foodMethods.addActivity,
            payload:{
                activity:dummy_sport
            }
        }

        const prevState = macroTracker(initState,action);

        action.type = foodMethods.resetDiary;
        expect(macroTracker(prevState,action)).toMatchSnapshot()
    
    });

})