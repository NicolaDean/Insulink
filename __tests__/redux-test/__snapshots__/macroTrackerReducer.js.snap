// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESTING THE REDUX REDUCER FOR MACRO: Add,Edit,Remove Food 1`] = `
Object {
  "activities": Object {
    "breakfast": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "dinner": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "lunch": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "snack": Object {
      "sports": Array [],
      "totCal": 0,
    },
  },
  "activity_id": 1,
  "currentDate": "[TODAY DATE]",
  "currentMeal": "breakfast",
  "history": Object {
    "activities": Object {
      "breakfast": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "dinner": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "lunch": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "snack": Object {
        "sports": Array [],
        "totCal": 0,
      },
    },
    "empty": true,
    "meals": Object {
      "breakfast": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "dinner": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "lunch": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "snack": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
    },
    "totMacro": Object {
      "cal": 0,
      "carb": 0,
      "fat": 0,
      "prot": 0,
    },
  },
  "id": 2,
  "meals": Object {
    "breakfast": Object {
      "foods": Array [
        Object {
          "cal": 90,
          "carb": 33,
          "fat": 55,
          "id": Object {
            "common_type": null,
            "food_name": "apple pie",
            "locale": "en_US",
            "photo": Object {
              "thumb": "https://nix-tag-images.s3.amazonaws.com/1291_thumb.jpg",
            },
            "serving_qty": 1,
            "serving_unit": "piece (1/8 of 9\\" dia)",
            "tag_id": "15060",
            "tag_name": "Apple pie",
          },
          "identifier": 1,
          "image": "[IMAGE]",
          "name": "apple",
          "prot": 22,
          "quantity": 1,
          "unit": "g",
        },
      ],
      "macro": Object {
        "cal": 90,
        "carb": 33,
        "fat": 55,
        "prot": 22,
      },
    },
    "dinner": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "lunch": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "snack": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
  },
  "totCalBurned": 0,
  "totMacro": Object {
    "cal": 90,
    "carb": 33,
    "fat": 55,
    "prot": 22,
  },
}
`;

exports[`TESTING THE REDUX REDUCER FOR MACRO: Add,Edit,Remove Food 2`] = `
Object {
  "activities": Object {
    "breakfast": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "dinner": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "lunch": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "snack": Object {
      "sports": Array [],
      "totCal": 0,
    },
  },
  "activity_id": 1,
  "currentDate": "[TODAY DATE]",
  "currentMeal": "breakfast",
  "history": Object {
    "activities": Object {
      "breakfast": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "dinner": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "lunch": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "snack": Object {
        "sports": Array [],
        "totCal": 0,
      },
    },
    "empty": true,
    "meals": Object {
      "breakfast": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "dinner": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "lunch": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "snack": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
    },
    "totMacro": Object {
      "cal": 0,
      "carb": 0,
      "fat": 0,
      "prot": 0,
    },
  },
  "id": 2,
  "meals": Object {
    "breakfast": Object {
      "foods": Array [
        Object {
          "cal": 180,
          "carb": 55,
          "fat": 33,
          "id": Object {
            "common_type": null,
            "food_name": "apple pie",
            "locale": "en_US",
            "photo": Object {
              "thumb": "https://nix-tag-images.s3.amazonaws.com/1291_thumb.jpg",
            },
            "serving_qty": 1,
            "serving_unit": "piece (1/8 of 9\\" dia)",
            "tag_id": "15060",
            "tag_name": "Apple pie",
          },
          "identifier": 1,
          "image": "[IMAGE]",
          "name": "apple",
          "prot": 77,
          "quantity": 2,
          "unit": "g",
        },
      ],
      "macro": Object {
        "cal": 180,
        "carb": 55,
        "fat": 33,
        "prot": 77,
      },
    },
    "dinner": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "lunch": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "snack": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
  },
  "totCalBurned": 0,
  "totMacro": Object {
    "cal": 180,
    "carb": 55,
    "fat": 33,
    "prot": 77,
  },
}
`;

exports[`TESTING THE REDUX REDUCER FOR MACRO: Add,Edit,Remove Food 3`] = `
Object {
  "activities": Object {
    "breakfast": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "dinner": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "lunch": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "snack": Object {
      "sports": Array [],
      "totCal": 0,
    },
  },
  "activity_id": 1,
  "currentDate": "[TODAY DATE]",
  "currentMeal": "breakfast",
  "history": Object {
    "activities": Object {
      "breakfast": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "dinner": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "lunch": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "snack": Object {
        "sports": Array [],
        "totCal": 0,
      },
    },
    "empty": true,
    "meals": Object {
      "breakfast": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "dinner": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "lunch": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "snack": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
    },
    "totMacro": Object {
      "cal": 0,
      "carb": 0,
      "fat": 0,
      "prot": 0,
    },
  },
  "id": 2,
  "meals": Object {
    "breakfast": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "dinner": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "lunch": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "snack": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
  },
  "totCalBurned": 0,
  "totMacro": Object {
    "cal": 0,
    "carb": 0,
    "fat": 0,
    "prot": 0,
  },
}
`;

exports[`TESTING THE REDUX REDUCER FOR MACRO: Add,Remove  Sport 1`] = `
Object {
  "activities": Object {
    "breakfast": Object {
      "sports": Array [
        Object {
          "calories": 400,
          "duration": 50,
          "identifier": 1,
          "name": "Nome",
          "photo": "[IMAGE]",
        },
      ],
      "totCal": 400,
    },
    "dinner": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "lunch": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "snack": Object {
      "sports": Array [],
      "totCal": 0,
    },
  },
  "activity_id": 2,
  "currentDate": Any<String>,
  "currentMeal": "breakfast",
  "history": Object {
    "activities": Object {
      "breakfast": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "dinner": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "lunch": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "snack": Object {
        "sports": Array [],
        "totCal": 0,
      },
    },
    "empty": true,
    "meals": Object {
      "breakfast": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "dinner": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "lunch": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "snack": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
    },
    "totMacro": Object {
      "cal": 0,
      "carb": 0,
      "fat": 0,
      "prot": 0,
    },
  },
  "id": 1,
  "meals": Object {
    "breakfast": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "dinner": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "lunch": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "snack": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
  },
  "totCalBurned": 400,
  "totMacro": Object {
    "cal": 0,
    "carb": 0,
    "fat": 0,
    "prot": 0,
  },
}
`;

exports[`TESTING THE REDUX REDUCER FOR MACRO: Add,Remove  Sport 2`] = `
Object {
  "activities": Object {
    "breakfast": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "dinner": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "lunch": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "snack": Object {
      "sports": Array [],
      "totCal": 0,
    },
  },
  "activity_id": 2,
  "currentDate": "[TODAY DATE]",
  "currentMeal": "breakfast",
  "history": Object {
    "activities": Object {
      "breakfast": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "dinner": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "lunch": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "snack": Object {
        "sports": Array [],
        "totCal": 0,
      },
    },
    "empty": true,
    "meals": Object {
      "breakfast": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "dinner": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "lunch": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "snack": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
    },
    "totMacro": Object {
      "cal": 0,
      "carb": 0,
      "fat": 0,
      "prot": 0,
    },
  },
  "id": 1,
  "meals": Object {
    "breakfast": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "dinner": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "lunch": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "snack": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
  },
  "totCalBurned": 0,
  "totMacro": Object {
    "cal": 0,
    "carb": 0,
    "fat": 0,
    "prot": 0,
  },
}
`;

exports[`TESTING THE REDUX REDUCER FOR MACRO: Reset Diary 1`] = `
Object {
  "activities": Object {
    "breakfast": Object {
      "sports": Array [
        Object {
          "calories": 400,
          "duration": 50,
          "identifier": 1,
          "name": "Nome",
          "photo": "[IMAGE]",
        },
      ],
      "totCal": 400,
    },
    "dinner": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "lunch": Object {
      "sports": Array [],
      "totCal": 0,
    },
    "snack": Object {
      "sports": Array [],
      "totCal": 0,
    },
  },
  "activity_id": 1,
  "currentDate": undefined,
  "currentMeal": "breakfast",
  "history": Object {
    "activities": Object {
      "breakfast": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "dinner": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "lunch": Object {
        "sports": Array [],
        "totCal": 0,
      },
      "snack": Object {
        "sports": Array [],
        "totCal": 0,
      },
    },
    "empty": true,
    "meals": Object {
      "breakfast": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "dinner": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "lunch": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
      "snack": Object {
        "foods": Array [],
        "macro": Object {
          "cal": 0,
          "carb": 0,
          "fat": 0,
          "prot": 0,
        },
      },
    },
    "totMacro": Object {
      "cal": 0,
      "carb": 0,
      "fat": 0,
      "prot": 0,
    },
  },
  "id": 1,
  "meals": Object {
    "breakfast": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "dinner": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "lunch": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
    "snack": Object {
      "foods": Array [],
      "macro": Object {
        "cal": 0,
        "carb": 0,
        "fat": 0,
        "prot": 0,
      },
    },
  },
  "totCalBurned": 0,
  "totMacro": Object {
    "cal": 0,
    "carb": 0,
    "fat": 0,
    "prot": 0,
  },
}
`;
